07-practical-work-1/COURSE_MATERIAL.md

```diff
@@ -75,82 +80,94 @@ To announce your group, create a new GitHub Discussion at
+| 12  | The repository contains meaningful example files to allow new users/developers (such as the teaching staff) to try out your application locally - these files can be the same as the ones used in the examples and outputs snippets |    0.2 |
+
+### Category 3 - Java, IntelliJ IDEA and Maven
+
+| #   | Criterion                                                                                                                                                                                          | Points |
+| --- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -----: |
+| 13  | The codebase is well structured, easy to access, easy to understand and is documented so it is easier for new comers to understand the codebase                                                    |    0.2 |
+| 14  | The codebase contains the two IntelliJ IDEA configuration files (_"Run the application"_ and _"Package application as JAR file"_) so developers can run and build the application within their IDE |    0.2 |
+| 15  | The codebase contains the Maven wrapper configuration file and scripts so developers can build the application without an IDE and without having to install Maven                                  |    0.2 |
+| 16  | The codebase is built with Maven and outputs an executable JAR file so the application can be ran everywhere Java is installed                                                                     |    0.2 |
+
+### Category 4 - Java IOs
+
+| #   | Criterion                                                                                                                                                                        | Points |
+| --- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -----: |
+| 17  | The CLI displays a comprehensive help message on how to use the application and displays errors on invalid/missing inputs and/or processing errors                               |    0.2 |
+| 18  | The CLI takes an input file and an output file as mandatory arguments and other optional arguments for customization (such as the input file/output file character encodings)    |    0.2 |
+| 19  | The CLI processes the files with buffers so it stays efficient even with large files                                                                                             |    0.2 |
+| 20  | The CLI processes the files respecting their respective charset(s)/character encoding(s) to ensure readability between operating systems/languages                               |    0.2 |
+| 21  | The CLI correctly manages resources with try-with-resources or try-catch-finally blocks to cleanly manage allocated resources in case a problem occurs when processing the files |    0.2 |
```

Comment:
These criteria seem fine to me, but wouldn't it be better to let students make their own choices ? The idea is to see if the theory has been acquired, so I won't be giving so much help. I'd use more generic and vague criteria

Reply:
You're right.